src/lib.rs

```diff
@@ -348,6 +352,92 @@ pub(crate) trait Castable {
     type RustType;
 }
 
+/// Defines a new [`Castable`] opaque struct with [`OwnershipBox`] ownership.
+///
+/// Expects to be invoked with a visibility specifier, the struct keyword, a struct name, and
+/// in parens, the Rust type that the opaque struct wraps. Similar to defining a normal struct
+/// that wraps a single type as its `.0` member.
+macro_rules! box_castable {
+    (
+        $(#[$comment:meta])*
+        $enum_vis:vis struct $name:ident($rust_type:ty);
+    ) => {
+        crate::castable!(OwnershipBox $(#[$comment])* $enum_vis $name $rust_type);
+    };
+}
+
+pub(crate) use box_castable;
+
+/// Defines a new [`Castable`] opaque struct with [`OwnershipArc`] ownership.
+///
+/// Expects to be invoked with a visibility specifier, the struct keyword, a struct name, and
+/// in parens, the Rust type that the opaque struct wraps. Similar to defining a normal struct
+/// that wraps a single type as its `.0` member.
+macro_rules! arc_castable {
+    (
+        $(#[$comment:meta])*
+        $enum_vis:vis struct $name:ident($rust_type:ty);
+    ) => {
+        crate::castable!(OwnershipArc $(#[$comment])* $enum_vis $name $rust_type);
+    };
+}
+
+pub(crate) use arc_castable;
+
+/// Defines a new [`Castable`] opaque struct with [`OwnershipArc`] ownership.
+///
+/// Expects to be invoked with a visibility specifier, the struct keyword, a struct name, and
+/// in parens, the Rust type that the opaque struct wraps (optionally with a lifetime specifier).
+/// Similar to defining a normal struct that wraps a single type as its `.0` member.
+///
+/// If a lifetime is specified, the opaque struct will be parameterized by that lifetime
+/// and a `PhantomData` field referencing the lifetime is added to the struct.
```

Comment:
```suggestion
/// If a lifetime parameter is specified, the opaque struct will be parameterized it
/// and a `PhantomData` field referencing the lifetime is added to the struct.
```

Nitpicking but I think it's important to be clear that it's a lifetime parameter rather than a specific lifetime (which can't be named).

Reply:
That's a good clarification. Applied.